lib/posix/shell/Kconfig

```diff
@@ -7,6 +7,7 @@ if SHELL
 
 config POSIX_SHELL
 	bool
+	default y if SHELL_ALL
```

Comment:
This should be removed as well.

Note: this is a non-user-configurable (i.e. it is a common component selected by other Kconfig options).